matscipy/gamma_surface.py

```diff
@@ -0,0 +1,699 @@
+                curr_ax.set_ylabel(self._vec_to_miller(self.surf_directions[ydir]))
+
+
+                if CNA_color:
+                    plot_atoms(atoms, ax=curr_ax, colors=atoms.get_array("colors"), 
+                    rotation=rot, **kwargs)
+                else: # default color are jmol (same is nglview)
+                    plot_atoms(atoms, ax=curr_ax, rotation=rot, **kwargs)
+                # avoid changing the size of the plot during animation
+                if keep_lims:
+                    curr_ax.set_xlim(xlim)
+                    curr_ax.set_ylim(ylim)
+
+
+        if plot_energies:
+            if self.Es is not None:
+                Es = self.Es
+            else:
+                try:
+                    # Assume that calculator is attached so we can get energy densities
+                    Es = self.get_energy_densities()
+                except:
+                    raise RuntimeError("Cannot plot energy densities before get_energy_densities is called!")
+                
+            self.plot_energy_densities(ax=energy_ax, si=si)
+        else:
+            energy_ax.clear()
+            energy_ax.axis("off")
+
+        plt.tight_layout()
+
+        plot_all_atom_axes(images[-1], keep_lims=False)
+
+        def drawimage(framedata):
+            framenum, atoms = framedata
+            # Plot Structure
+            
+            plot_all_atom_axes(atoms, keep_lims=True)
+
+            # Plot energies
+            if plot_energies:
+                # Manually clear the colorbar
+                energy_ax.images[-1].colorbar.remove()
+                energy_ax.clear()
+                self.plot_energy_densities(ax=energy_ax, si=si)
+                pos = self.offsets[framenum]
+                plt.scatter(pos[0], pos[1], marker="x", color="k")
+
+
+        animation = FuncAnimation(fig, drawimage, frames=enumerate(images),
+                                init_func=lambda: None,
+                                interval=200)
+        return animation
+
+
+class StackingFault(GammaSurface):
+    '''
+    Class for stacking fault-specific generation & plotting
+    '''
+
+    def generate_images(self, n, *args, **kwargs):
+        '''
+        Generate gamma surface images on an (n) line
+        n: int
+            Number of images
+        '''
+        return super().generate_images(1, n, *args, **kwargs)
+    
+    def plot_energy_densities(self, Es=None, ax=None, si=False):
+        '''
+        Produce a matplotlib plot of the stacking fault energy, from the data gathered in self.generate_images and self.get_surface_energy
+
+        Returns a matplotlib fig and ax object
+        Es: np.array
+            (nx, ny) array of energy densities. If None, uses self.Es (if populated from self.get_energy_densities())
+        ax: matplotlib axis object
+            Axis to draw plot
+        si: bool
+            Use SI units (J/m^2) if True, else atomic units (eV/A^2)
```

Comment:
eV/A^2 are ASE units, not atomic units (which would be Hartree/bohr^2 in this case)